File: unit-3/sprint-2/day-1/assignments/Play Some Music/app/src/main/java/com/example/playsomemusic/MusicService.java
package com.example.playsomemusic;

import android.app.Notification;
import android.app.NotificationChannel;
import android.app.NotificationManager;
import android.app.Service;
import android.content.Context;
import android.content.Intent;
import android.graphics.Color;
import android.media.MediaPlayer;
import android.os.Build;
import android.os.IBinder;

import androidx.annotation.RequiresApi;
import androidx.core.app.NotificationCompat;

public class MusicService extends Service {

    private MediaPlayer mediaPlayer;

    public MusicService() {
    }

    @Override
    public IBinder onBind(Intent intent) {
        // TODO: Return the communication channel to the service.
        throw new UnsupportedOperationException("Not yet implemented");
    }

    @Override
    public void onCreate() {
        super.onCreate();
        mediaPlayer = MediaPlayer.create(this,R.raw.songs);

    }

    @Override
    public void onDestroy() {
        super.onDestroy();
        mediaPlayer.stop();
    }

    @RequiresApi(api = Build.VERSION_CODES.O)
    @Override
    public int onStartCommand(Intent intent, int flags, int startId) {
        startMusic();
        showNotificationAndStartForeGround();
        return super.onStartCommand(intent, flags, startId);
    }

    private void startMusic() {
        Thread thread = new Thread(task);
        thread.start();
    }

    private Runnable task = new Runnable() {
        @Override
        public void run() {
            mediaPlayer.start();
        }
    };

    @RequiresApi(api = Build.VERSION_CODES.O)
    private void showNotificationAndStartForeGround() {
        String NOTIFICATION_CHANNEL_ID = "Sudarshan";
        String channelName = "Background Service";
        NotificationChannel chan = new NotificationChannel(NOTIFICATION_CHANNEL_ID, channelName, NotificationManager.IMPORTANCE_NONE);
        chan.setLightColor(Color.BLUE);

        NotificationManager manager = (NotificationManager) getSystemService(Context.NOTIFICATION_SERVICE);
        assert manager != null;
        manager.createNotificationChannel(chan);

        NotificationCompat.Builder notificationBuilder = new NotificationCompat.Builder(this, NOTIFICATION_CHANNEL_ID);
        Notification notification = notificationBuilder.setOngoing(true)
                .setContentTitle("app is working fine ")
                .setContentText("Hey music is playing")
                .setPriority(NotificationManager.IMPORTANCE_MIN)
                .setCategory(Notification.CATEGORY_SERVICE)
                .build();
        startForeground(2, notification);
    }

}

File: unit-3/sprint-4/evaluation/MusicPlayer/app/src/main/java/com/example/musicplayer/MusicService.kt
package com.example.musicplayer

import android.app.Notification
import android.app.NotificationChannel
import android.app.NotificationManager
import android.app.Service
import android.content.Context
import android.content.Intent
import android.graphics.Color
import android.media.MediaPlayer
import android.os.Build
import android.os.IBinder
import androidx.annotation.RequiresApi
import androidx.core.app.NotificationCompat


class MusicService() : Service() {


    private var mediaPlayer: MediaPlayer? = null

    override fun onCreate() {
        super.onCreate()
        mediaPlayer = MediaPlayer.create(this,R.raw.songs)
    }

    override fun onStartCommand(intent: Intent?, flags: Int, startId: Int): Int {
        startMusic()
        return super.onStartCommand(intent, flags, startId)
    }



    private fun startMusic() {
        val thread = Thread(task)
        thread.start()
    }

    private val task = Runnable {
        mediaPlayer!!.start()
    }

    override fun onBind(intent: Intent?): IBinder? {
        TODO("Not yet implemented")
    }

    fun play() {
        if (!mediaPlayer!!.isPlaying) mediaPlayer!!.start()
    }

    fun pause() {
        if (mediaPlayer!!.isPlaying) mediaPlayer!!.pause()
    }

}